paperscraper/pdf.py

```diff
@@ -105,7 +106,7 @@ def save_pdf(
     else:  # if no citation_pdf_url meta tag found, try other fallbacks
```

Comment:
Add fallback if everything passes at first, but there is an issue with PDF integrity. (Happened with the PMC DOI "10.1038/s42256-023-00639-z". This code change is required for the functional "no api key" tests to pass.
```suggestion
            logger.info("Attempting download via fallback mechanisms instead")
            success = fallback_bioc_pmc(paper_metadata["doi"], output_path)
            if not success:
                # Check for specific publishers
                if "elife" in paper_metadata["doi"].lower():
                    logger.info("Attempting fallback to eLife XML repository")
                    fallback_elife_xml(paper_metadata["doi"], output_path)
                elif api_keys and "WILEY_TDM_API_TOKEN" in api_keys:
                    fallback_wiley_api(paper_metadata, output_path, api_keys)
                elif api_keys and "ELSEVIER_TDM_API_KEY" in api_keys:
                    fallback_elsevier_api(paper_metadata, output_path, api_keys)
    else:  # if no citation_pdf_url meta tag found, try other fallbacks
```

Reply:
Thanks good catch!